libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxVoiceInstructionsPlayer.kt

```diff
@@ -113,12 +128,52 @@ class MapboxVoiceInstructionsPlayer @JvmOverloads constructor(
         announcement: SpeechAnnouncement,
         consumer: MapboxNavigationConsumer<SpeechAnnouncement>
     ) {
-        playCallbackQueue.add(PlayCallback(announcement, consumer))
+        play(announcement, consumer, 500)
+    }
+
+    /**
+     * Given [SpeechAnnouncement] the method will play the voice instruction.
+     * If a voice instruction is already playing or other announcement are already queued,
+     * the given voice instruction will be queued to play after.
+     * @param announcement object including the announcement text
+     * and optionally a synthesized speech mp3.
+     * @param consumer represents that the speech player is done playing
+     * @param expirationMillis if the specified timeout passes between the moment the instruction
+     *  was removed from the queue to be played and the moment the playing actually starts,
+     *  it will be cancelled. As an alternative, you may manually invoke the [cancel] method
+     *  as soon as playing is no longer required. If null is provided, it will never be cancelled.
+     */
+    fun play(
+        announcement: SpeechAnnouncement,
+        consumer: MapboxNavigationConsumer<SpeechAnnouncement>,
+        expirationMillis: Long?
+    ) {
+        playCallbackQueue.add(PlayCallback(announcement, consumer, expirationMillis))
         if (playCallbackQueue.size == 1) {
             play()
         }
     }
 
+    /**
+     * Cancels playing the given announcement. Use in cases when playing the instruction
+     * after some time or after a certain event becomes useless.
+     * As an alternative, you may pass `expirationMillis` parameter to [play] method.
+     * It will cancel the playing automatically after the specified timeout.
+     *
+     * @param announcement the instruction to be cancelled
+     */
+    fun cancel(announcement: SpeechAnnouncement) {
```

Comment:
We already have the `clear()` method. What's the difference between `cancel()' and `clear()`?

Reply:
Because it's about this:

> Clears any announcements queued.

While here I want to expose a way to cancel a specific announcement.